### ISSUE TITLE
The currently focused window receives the key before the change

### ISSUE THREAD

**Issue by josegrabelha** [agabor/RightKeyboard#3]
Hello,
Like the original version, "when the program switches the input language, the currently focused window receives the key before the change. Because of that, the first key that is pressed on another keyboard is always incorrect, unless that particular key is the same on the previous language." Can you please fix it ?
Thank you!

**Comment by auralarch** [comment]
Based on my understanding of how this works and "how windows works" (to put it simply)  This is unavoidable and only Microsoft could "fix" this.  However, this is already working outside of and beyond the intended behavior envisioned by Windows developers.

**Comment by tssajo** [comment]
What @auralarch says is 100% true.
As a workaround, when switching to a different keyboard I always hit the Shift key once, that makes the actual layout switch. After that you can keep typing and all keys hit after Shift will be using the correct keyboard layout.

**Comment by vnathalye** [comment]
Yes. I agree with @tssajo. For that matter, you can press any non-printing key (shift, ctrl, alt, esc, arrow keys) to activate that keyboard and then you won't have incorrect letter typed. I've started using this code yesterday and I'm planning to make it a habit to use either right alt / right ctrl as the activation key. After all these keys are seldom used by me and these keys, kind of, match the application name RightKeyboard too :)
